src/analysis/rust_type.rs

```diff
@@ -1,49 +1,152 @@
+use std::result;
+
 use library;
 
+pub struct Result(result::Result<String, String>);
```

Comment:
A trait makes more sense to me -- you just import it in modules where this sugar is needed instead of paying the price everywhere such a `Result` is created.